packages/@aws-cdk/aws-servicecatalog/lib/tag-options.ts

```diff
@@ -1,14 +1,56 @@
+import * as cdk from '@aws-cdk/core';
+import { hashValues } from './private/util';
+import { InputValidator } from './private/validation';
+import { CfnTagOption } from './servicecatalog.generated';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from 'constructs';
+
 /**
- * Defines a Tag Option, which are similar to tags
- * but have multiple values per key.
+ * Properties for TagOptions.
  */
-export class TagOptions {
+export interface TagOptionsProps {
   /**
-  * List of CfnTagOption
-  */
-  public readonly tagOptionsMap: { [key: string]: string[] };
+   * TagOption keys and associated list of possible values.
+   */
+  readonly tagOptions: { [key: string]: string[] };
+}
+
+/**
+ * Defines a set of TagOptions, which are a list of key-value pairs managed in AWS Service Catalog.
+ * It is not an AWS tag, but serves as a template for creating an AWS tag based on the TagOption.
+ * @resource AWS::ServiceCatalog::TagOption
+ */
+export class TagOptions extends cdk.Resource {
+  /**
+   * Map of underlying TagOption resources.
+   *
+   * @internal
+   */
+  public readonly _tagOptionsMap: { [key: string]: CfnTagOption };
+
+  constructor(scope: Construct, id: string, props: TagOptionsProps) {
+    super(scope, id);
+    this._tagOptionsMap = this.createUnderlyingTagOptions(props.tagOptions);
+  }
 
-  constructor(tagOptionsMap: { [key: string]: string[]} ) {
-    this.tagOptionsMap = { ...tagOptionsMap };
+  private createUnderlyingTagOptions(tagOptions: { [key: string]: string[] }): { [key: string]: CfnTagOption } {
+    var tagOptionMap: { [key: string]: CfnTagOption } = {};
+    for (const [key, tagOptionsList] of Object.entries(tagOptions)) {
+      InputValidator.validateLength(this.node.addr, 'TagOption key', 1, 128, key);
+      tagOptionsList.forEach((value: string) => {
+        InputValidator.validateLength(this.node.addr, 'TagOption value', 1, 256, value);
+        const tagOptionIdentifier = `$TagOption${hashValues(key, value)}`;
+        if (!this.node.tryFindChild(tagOptionIdentifier)) {
```

Comment:
This is carryover from how we originally did things so that if you for example had duplicate values it would be idempotent and ignore.  I would prefer it if we could modify it so that creation is very strict, but that would change behavior again.